Mage/src/main/java/mage/game/combat/CombatGroup.java

```diff
@@ -720,4 +792,26 @@ public boolean changeDefenderPostDeclaration(UUID newDefenderId, Game game) {
         }
         return false;
     }
+
+    public boolean defenderControlsDefensiveFormation(Game game) {
+        // for handling Defensive Formation
+        for (Permanent defensiveFormation : game.getBattlefield().getAllActivePermanents(defendingPlayerId)) {
+            if (defensiveFormation.getAbilities().containsKey(ControllerAssignCombatDamageToBlockersAbility.getInstance().getId())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean isButcherOrgg(Permanent creature, UUID playerId, boolean first, Game game, boolean isAttacking) {
```

Comment:
Why are you using card's hacks in main code? It's a BAD thing. Use normal names for your checks. That code will use for other cards, not only Butcher Orgg. And add tests for it. I think you should approve your code with @LevelX2 -- I have bad feelings about it.

Reply:
I believed it could be acceptable here since Butcher Orgg is the only creature in the game using this form of damage distribution (unless something similar comes out in the future). Similar case with Defensive Formation - of course, I see no problem changing the name, especially if the Defensive Formation code is possible to reuse for Banding somehow.